### PR TITLE
fix: JS library not found in view mode after importing from Git

### LINKED ISSUE #29278: The custom JS library is not present right after importing an Application
Steps to reproduce:
---------------------
1. Create an application and add a custom JS library
2. Publish the application
3. Connect to git and make sure application is pushed to git
4. Import the application as a new application from Git
5. Open the deployed version of the application. The custom JS library will not be present in deployed version.

### PR DESCRIPTION
**Pull request by nayan-rafiq**
## Description
If an application has a custom js library set, it's not available in view mode right after importing the application from git. User need to add a commit to make it available in Git. This PR performs a publish application operation right after git import so that any resource available in edit mode is also available in view mode.

#### PR fixes following issue(s)
Fixes #29278

#### Type of change
- Bug fix (non-breaking change which fixes an issue)

#### How Has This Been Tested?
- [ ] Manual
- [ ] JUnit

#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **New Features**
  - Applications can now be automatically published after being imported from Git.

- **Tests**
  - Updated tests to include new assertions for verifying the behavior of unpublished custom JavaScript libraries.

### CODE REVIEW

File: app/server/appsmith-server/src/test/java/com/appsmith/server/services/ce/GitServiceCETest.java [hunk 1]
```diff
@@ -3498,6 +3498,8 @@ public void importApplicationFromGit_validRequest_Success() {
                                     .getGitAuth()
                                     .getPublicKey())
                             .isEqualTo(gitAuth.getPublicKey());
+                    assertThat(application.getUnpublishedCustomJSLibs().size())
+                            .isEqualTo(application.getUnpublishedCustomJSLibs().size());
```
[the remark this answers is not in the document]
Comment (nayan-rafiq) [reply app/server/appsmith-server/src/test/java/com/appsmith/server/services/ce/GitServiceCETest.java]: @coderabbitai review

[the remark this answers is not in the document]
Comment (nayan-rafiq) [reply app/server/appsmith-server/src/test/java/com/appsmith/server/services/ce/GitServiceCETest.java]: @coderabbitai resolve

**Review by AnaghHegde — APPROVED** [review]

**Pull request merged**